### ISSUE TITLE
Port twisted.internet.protocol to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6060]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6060|
|Type|enhancement|
|Created|2012-10-04 18:34:40Z|
|Branch|https://github.com/twisted/twisted/tree/protocol-py3-6060|

`twisted.internet.protocol` should run on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6060 6060
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_protocol_py3_6060 branches/protocol-py3-6060
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349375680000000 1349375680000000
changetime__1349803148000000 1349803148000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35952]) Branching to 'protocol-py3-6060'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

I improved test coverage somewhat, and moved over a couple of tests from `twisted.test.test_protocols` to a better location (they were otherwise unchanged). I opened a separate ticket ([#6064](https://github.com/twisted/twisted/issues/6064)) for FileWrapper test coverage.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/protocol-py3-6060 has been started.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

Oh, and I should note I didn't do tests for the datagram and process classes since they're (somewhat) tested by UDP and process code.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

Woo!  Tests yay.  Thanks.

  1. `__future__` imports in `twisted.internet.protocol`
  1. `AdapterTests.test_consumerToProtocol` docstring says `IConsumer` where it means `IProtocol`
  1. You're also marked as the maintainer of `twisted.internet.protocol`, in case you wanted to do something about that. :)

That's all.  Please merge when these are addressed.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [36029]) Merge protocol-py3-6008.

Author: itamar
Review: exarkun
Fixes: [#6060](https://github.com/twisted/twisted/issues/6060)

Port twisted.internet.protocol to Python 3.
